tests/suite/graceful_recovery_test.go

```diff
@@ -60,6 +62,12 @@ var _ = Describe("Graceful Recovery test", Ordered, Label("nfr", "graceful-recov
 		Expect(podNames).To(HaveLen(1))
 
 		ngfPodName = podNames[0]
+		if portFwdPort != 0 {
+			coffeeURL = fmt.Sprintf("%s:%d/coffee", baseHttpURL, portFwdPort)
+		}
+		if portFwdHTTPSPort != 0 {
+			teaURL = fmt.Sprintf("%s:%d/tea", baseHttpsURL, portFwdHTTPSPort)
```

Comment:
I see this error in the pipeline even though the test passes

```
E0603 18:20:27.949715   20071 portforward.go:413] an error occurred forwarding 10443 -> 443: error forwarding port 443 to pod 7670f03f4c4311a5e5e01c2bc6a68ac91139d329cc7e10669bc8703b7887f24a, uid : failed to execute portforward in network namespace "/var/run/netns/cni-2649e88b-23b3-169b-1879-bff2bac7f7e5": failed to connect to localhost:443 inside namespace "7670f03f4c4311a5e5e01c2bc6a68ac91139d329cc7e10669bc8703b7887f24a", IPv4: dial tcp4 127.0.0.1:443: connect: connection refused IPv6 dial tcp6 [::1]:443: connect: connection refused 
2024/06/03 18:20:27 ERROR error forwarding ports error="lost connection to pod"
2024/06/03 18:20:27 INFO retrying port forward in 100ms...
```

I did a quick search and could it possibly be that the tea service is still listening on port 80? Like nothing actually listens to the portFwdHTTPSPort.

Reply:
I checked using curl requests and `tea` is listening on 10443. The error is maybe cause it tried port-forwarding and fails and succeeds on retry

Reply:
Hmm, ok yea that sounds plausible